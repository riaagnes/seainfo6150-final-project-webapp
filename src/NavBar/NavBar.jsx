import React, { useState } from "react";
import { Switch, Route, Link } from "react-router-dom";
import styles from "./NavBar.module.css";
import "../App.css";
import Form from "../Contactus/Form";
import Home from "../Home/Home";
import AboutUs from "../AboutUs/AboutUs.jsx";
import IndividualMountain from "../IndividualMountain/IndividualMountain";
import Error from "../Error/Error.jsx";
const NavBar = (props) => {
  const [navLinkToggle, setNavLinkToggle] = useState(false);
  const [className, setClassName] = useState("navlinks");
  const handleNavLinksToggle = () => {
    setNavLinkToggle(!navLinkToggle);
    if (navLinkToggle) {
      setClassName("navlinks active");
    } else {
      setClassName("navlinks");
    }
  };

  return (
    <div>
      <nav>
        <div className={styles.logo}>
          <i className="fas fa-mountain mon"></i>
          <h3>SeaMountains</h3>
        </div>
        <ul className={className}>
          <li className={styles.link}>
            <Link to="/">Home</Link>
          </li>
          <li className={styles.link}>
            <Link to="/contact">Contact us</Link>
          </li>
          <li className={styles.link}>
            <Link to="/about">About</Link>
          </li>
        </ul>
        <div onClick={handleNavLinksToggle} className={styles.hamburgerToggle}>
          <i className="fas fa-bars fa-lg"></i>
        </div>
      </nav>

      <Switch>
        <Route path="/" exact render={() => <Home mountains={props} />} />
        <Route path="/contact" exact render={() => <Form />} />
        <Route path="/about" exact render={() => <AboutUs />} />

        <Route
          exact
          path={`/mountainlist/:id`}
          render={({ match }) => {
            // getting the parameters from the url and passing
            // down to the component as props

            return props ? (
              <IndividualMountain mountain={props.mountains[match.params.id]} />
            ) : null;
          }}
        />
        <Route component={Error} />
      </Switch>
    </div>
  );
};
export default NavBar;
