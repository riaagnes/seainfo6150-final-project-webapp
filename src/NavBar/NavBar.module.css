

nav{
    background: radial-gradient(at center center, #2d7e9270 0%, #5c7e9ec7 100%), url(../images/photo-1513496497-82958929944d.jpeg);
    background-size:cover;
    background-position: center;
    display: flex;
    justify-content: space-around;
    min-height: 15vh;
    /* position: absolute; */
    width: 100%;
    align-items: center;
   
}
.navlinks{
    display: flex;
    width:50%;
    justify-content: space-around;
    list-style: none;
    font-size: 24px;
}
.logo{
display: flex;
font-size: 32px;
color: rgba(255,255,255,.8)
}
.logo h3{
    margin-left: 7px;
    letter-spacing: 1px;
    font-weight: lighter;
}

.link{
letter-spacing: 3px;
font-weight: bolder;

}
.link a{
    color:rgba(255,255,255,.8);
    text-decoration: none;
}
.link a:hover{
    color:rgba(255,255,255,1);
}
.hamburgerToggle{
    cursor: pointer;
    color: white;
    display: none;
}
@media screen and (max-width: 850px){
    body{
        overflow-x: hidden;
    }
    .hamburgerToggle{
        display: block;
    }
  
    .navlinks{
        position: absolute;
        right: 0px;
        display: flex;
        flex-direction: column;
        height: 97vh;
        align-items: center;
        width: 100%;
        top:15vh;
        font-size: 20px;
        background: teal;
        transform: translateX(100%);
        color:black
    }
    .active{
        transform: translateX(0%);
        width:100%
    }
}