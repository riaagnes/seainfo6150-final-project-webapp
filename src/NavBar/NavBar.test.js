import React from "react";
import NavBar from "./NavBar";

describe("NavBar tests", () => {
    const props = {
      "mountains" : [
        {
          "id": "7",
          "name": "Mount Pilchuck",
          "height": "5327 feet",
          "image": {
              "_height": "619",
              "_url": "https://www.wta.org/site_images/hikes/mount-pilchuck-by-day-hike.jpg/@@images/45b388b1-1ef8-4a37-a017-9b0a25a137b8.jpeg",
              "_width": "1100"
          },
          "text": "Mount Pilchuck is a popular hiking destination for everyone from intermediate to experienced hikers. You may have also heard this mountain's name in association with local glass artist, Dale Chihuly, who founded a glassblowing school in the area",
          "details": {
              "image_url": "https://www.wta.org/site_images/hikes/mount-pilchuck-by-day-hike.jpg/@@images/45b388b1-1ef8-4a37-a017-9b0a25a137b8.jpeg",
              "detail": "Despite its intermediate difficulty, Mount Pilchuck sees a slew of visitors in a given year, as hikers flock to the trail for its grand panoramic views and historic restored fire lookout. Easily one of the most rewarding hikes in the area, its relatively short 2.7 mile, 2,300 feet ascent gifts you 360-degree views of Mount Baker, Mount Rainier and the Olympics from its shelter lookout, which sits nestled on a summit that tops out around 5,300 feet. This calf-friendly journey is all made possible by a high elevation trailhead, which sets you off on your travels around 3,000 feet, cutting what would otherwise be an arduous ten mile roundtrip in half. Don’t let Pilchuck’s popularity fool you, however. Even with its heavy foot traffic and respectable grade, this peak should not be mistaken for a bunny hill. There are perilous elements, especially in winter, when snow may obscure the path, leading hikers to improvise and become lost. Hikers often have to be rescued from the mountain, so please take all the necessary precautions before tackling the trail. It is always encouraged that you check in with the ranger and sign the registry at the trailhead before starting your journey. To begin, depart the trailhead and reach an immediate fork. The left trail is a false path, so steer right and get your feet wet crossing a stream, christening your journey up Mount Pilchuck. After about a quarter mile, you will see an “entering Pilchuck State Park” sign. It's seen better days, but it reaffirms your course and destination. There is sparse signage from this point on. It is not a particularly hazardous trail early on, but depending on the season, you could be dealing with rocky, snowy and wet conditions on the latter part of the trip, so be prepared for anything. Traction devices and poles are strongly encouraged if snow is present on the trail.You will encounter a scree slope at the one-mile mark (the elevation marker should be about 3,500 feet for reference). Do not attempt to scramble upward. Instead, look for an orange marker that guide you left through the boulder field and lead you back to the defined trail. After another half mile, you will depart the forest and reach a clearing. Unless cloud cover is obscuring it, the fire lookout should be visible at the summit. Use this landmark as reference, as hikers have been known to gravitate toward false summits and place themselves in danger. After snaking up the exposed slope, you will reach the cone of the summit. There should be a brown “Parking Area” sign on a tree pointing returning hikers back down the slope you have just hiked up. This will be the last vestige of signage before reaching the lookout. If you do not see this sign and are not confident of your whereabouts, it is possible that you have wandered toward one of the false summits. On the right trail, you will continue on, winding around the peak until you eventually reach the top of the mountain and its sublime restored fire lookout. Sit, eat and enjoy your time in the shelter. Originally constructed in 1921 (almost ten years after the first trail on the mountain was established), the fire lookout now stands today as a proud landmark for the area. The shelter has endured numerous renovations over the years and continues to be one of the most popular and in-demand lunch spots in Washington. Reservations are not necessary, however, as even on the busiest of days, spirits are high with summiters and the atmosphere is always welcoming and inclusive, even between large parties that camp out on the benches and outlying boulders. It’s hard not to find a smiling face on top of this picturesque peak. A social mountain, indeed. And one that should not be missed."
          }
      }
      ]
        
      
    } 
      
    

    
    it("renders correctly", () => {
      const { container } = render(<NavBar {...props}/>);
      expect(container).toMatchSnapshot();
    });
});