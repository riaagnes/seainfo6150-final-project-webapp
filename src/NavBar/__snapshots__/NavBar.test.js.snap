// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests renders correctly 1`] = `
<div>
  <div>
    <nav>
      <div
        class="logo"
      >
        <i
          class="fas fa-mountain mon"
        />
        <h3>
          SeaMountains
        </h3>
      </div>
      <ul
        class="navlinks"
      >
        <li
          class="link"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="link"
        >
          <a
            href="/contact"
          >
            Contact us
          </a>
        </li>
        <li
          class="link"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
      <div
        class="hamburgerToggle"
      >
        <i
          class="fas fa-bars fa-lg"
        />
      </div>
    </nav>
    <div>
      <ul
        class="container"
      >
        <li
          class="container"
        >
          <article
            class="article"
          >
            <header
              class="header"
            >
              <img
                class="image"
                src="https://www.wta.org/site_images/hikes/mount-pilchuck-by-day-hike.jpg/@@images/45b388b1-1ef8-4a37-a017-9b0a25a137b8.jpeg"
              />
              <h2
                class="title"
              >
                <a
                  class="link"
                  href="/mountainlist/7"
                >
                  Mount Pilchuck
                </a>
              </h2>
            </header>
            <div
              class="wrapper"
            />
            <button
              class="button"
            >
              Show more
            </button>
          </article>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
